content/technotes/2018-09-09-github-badges.md

```diff
@@ -0,0 +1,395 @@
+| CRAN                                                                                                       | <http://cran.r-project.org/package=drake>                | <http://www.r-pkg.org/badges/version/drake>                                           |
+| downloads                                                                                                  | <http://cran.rstudio.com/package=drake>                  | <http://cranlogs.r-pkg.org/badges/drake>                                              |
+| Codecov                                                                                                    | <https://codecov.io/github/ropensci/drake?branch=master> | <https://codecov.io/github/ropensci/drake/coverage.svg?branch=master>                 |
+| Zenodo                                                                                                     | <https://zenodo.org/badge/latestdoi/82609103>            | <https://zenodo.org/badge/82609103.svg>                                               |
+| Project Status: Active â€“ The project has reached a stable, usable state and is being actively developed. | <http://www.repostatus.org/#active>                      | <http://www.repostatus.org/badges/latest/active.svg>                                  |
+
+Quite handy for our metadata collection!
+
+Read the source for `extract_badges`
+[here](https://github.com/ropensci/codemetar/blob/dev/R/extract_badges.R)
+and see how it’s used in [this
+script](https://github.com/ropensci/codemetar/blob/dev/R/guess_metadata.R).
+You can also compare [`codemetar`’s
+README](https://github.com/ropensci/codemetar/blob/master/README.md)
+with its
+[codemeta.json](https://github.com/ropensci/codemetar/blob/master/codemeta.json)
+e.g. [the lines for
+“contIntegration”](https://github.com/ropensci/codemetar/blob/master/codemeta.json#L445).
+
+Now since, `codemetar::extract_badges` exports a nice data.frame for any
+README with badges, and is exported, it’d be too bad not to use it to
+gain insights from many, many READMEs!
+
+Extract badges from CRAN packages
+---------------------------------
+
+In this exploration we shall concentrate on CRAN packages that indicate
+a GitHub repo link under the URL field of DESCRIPTION. By the way, if
+you don’t indicate such links in DESCRIPTION of your package yet, you
+can (and should) run
+[`usethis::use_github_links`](http://usethis.r-lib.org/reference/use_github_links.html).
+
+### Get links to GitHub repos from CRAN information
+
+I reckon that I could have also used BugReports like Steven M Mortimer
+did [in his great analysis of CRAN downloads and GitHub
+stars](https://stevenmortimer.com/most-starred-r-packages-on-github/).
+Here’s how I got all the repo owners and names:
```

Comment:
This is cool!  What about use of badges from CRAN packages that don't have GitHub links?  Can we extract README addresses directly from CRAN? (I'm still a bit unclear why some CRAN packages don't have README links even when they do have READMEs on GitHub).  Anyway, a bit of a tangent.

Reply:
Good point! They wouldn't be Markdown though 🤔 will explore next week! At least as a remark.